contracts/strategies/HStrategy.sol

```diff
@@ -0,0 +1,841 @@
+        require(erc20Tokens.length == 2, ExceptionsLibrary.INVALID_LENGTH);
+        require(moneyTokens.length == 2, ExceptionsLibrary.INVALID_LENGTH);
+        require(uniV3Tokens.length == 2, ExceptionsLibrary.INVALID_LENGTH);
+        for (uint256 i = 0; i < 2; i++) {
+            require(erc20Tokens[i] == tokens_[i], ExceptionsLibrary.INVARIANT);
+            require(moneyTokens[i] == tokens_[i], ExceptionsLibrary.INVARIANT);
+            require(uniV3Tokens[i] == tokens_[i], ExceptionsLibrary.INVARIANT);
+        }
+        erc20Vault = erc20Vault_;
+        moneyVault = moneyVault_;
+        uniV3Vault = uniV3Vault_;
+        tokens = tokens_;
+        IUniswapV3Factory factory = IUniswapV3Factory(_positionManager.factory());
+        pool = IUniswapV3Pool(factory.getPool(tokens_[0], tokens_[1], fee_));
+        require(address(pool) != address(0), ExceptionsLibrary.ADDRESS_ZERO);
+    }
+
+    /// @notice creates the clone of the strategy
+    /// @param tokens_ the addresses of the tokens managed by the strategy
+    /// @param erc20Vault_ the address of the erc20 vault
+    /// @param moneyVault_ the address of the moneyVault. It is expected to be yEarn or AAVE
+    /// @param uniV3Vault_ the address of uniV3Vault. It is expected to not hold the position
+    /// @param fee_ the fee of the uniV3 pool on which the vault operates
+    /// @param admin_ the addres of the admin of the strategy
+    /// @return strategy the address of new strategy
+    function createStrategy(
+        address[] memory tokens_,
+        IERC20Vault erc20Vault_,
+        IIntegrationVault moneyVault_,
+        IUniV3Vault uniV3Vault_,
+        uint24 fee_,
+        address admin_
+    ) external returns (HStrategy strategy) {
+        strategy = HStrategy(Clones.clone(address(this)));
+        strategy.initialize(tokens_, erc20Vault_, moneyVault_, uniV3Vault_, fee_, admin_);
+    }
+
+    /// @notice updates parameters of the strategy. Can be called only by admin
+    /// @param newStrategyParams the new parameters
+    function updateStrategyParams(StrategyParams calldata newStrategyParams) external {
+        _requireAdmin();
+        int24 tickSpacing = pool.tickSpacing();
+        require(
+            newStrategyParams.halfOfShortInterval > 0 &&
+                (newStrategyParams.halfOfShortInterval % tickSpacing == 0) &&
+                newStrategyParams.tickNeighborhood <= newStrategyParams.halfOfShortInterval &&
+                newStrategyParams.tickNeighborhood >= 1000,
+            ExceptionsLibrary.INVARIANT
+        );
+
+        int24 globalIntervalWidth = newStrategyParams.domainUpperTick - newStrategyParams.domainLowerTick;
+        require(
+            (newStrategyParams.domainLowerTick % tickSpacing == 0) &&
+                (newStrategyParams.domainUpperTick % tickSpacing == 0) &&
+                globalIntervalWidth > 0 &&
+                globalIntervalWidth > newStrategyParams.halfOfShortInterval &&
```

Comment:
I think, that we have to check globalIntervalWidth > newStrategyParams.halfOfShortInterval * 2